### ISSUE TITLE
Remove dangling triangles after boolean operation

### ISSUE THREAD

**Issue by dhabbyc** [mikedh/trimesh#1925]
Hello,

 I have a mesh resulting from performing a boolean operation from two closed (volume) surfaces. But the result has dangling triangles (I suppose due to numeric coincidences). Is there a way to find all triangles that makes the resulting suface non-volume? I want to remove all faces that have at least one edge that has no connections to other edges.

Any help will be very much appreciated.

Regards.

**Comment by mikedh** [comment]
Hey, not sure if it'll help but you might take a look at `trimesh.repair.broken_faces`:

```

In [1]: import trimesh

In [2]: trimesh.repair.broken_faces?
Signature: trimesh.repair.broken_faces(mesh, color=None)
Docstring:
Return the index of faces in the mesh which break the
watertight status of the mesh.

Parameters
--------------
mesh : trimesh.Trimesh
  Mesh to check broken faces on
color: (4,) uint8 or None
  Will set broken faces to this color if not None

Returns
---------------
broken : (n, ) int
  Indexes of mesh.faces
File:      /media/psf/Home/Dropbox/robotics/trimesh/trimesh/repair.py
Type:      function

```

**Comment by dhabbyc** [comment]
Hello,

 Thank you for your help. Unfortunately, i've already tried using broken_faces. Also I have looked and tried using 'stitch'.

 I think the problem is on the boolean operation. I have two watertight volumes (checked by trimesh). But for reasons beyond me, I can't use boolean operations from trimesh (blender nor OpenSCAD). So the only thing that I have found for windows OS in Python is [pycork](https://github.com/drlukeparry/pycork). But the resulting meshes are 'weird' (non-volume, non-watertight), volumes (values) are almost equal though (blender/pycork).

Is there another alternative to perform boolean operations from Python on Windows OS?

Thank you again.

**Comment by mikedh** [comment]
Yeah boolean operations on meshes have always been pretty flaky in my experience. The only engine I've found that provides somewhat reliable results is OpenSCAD (i.e. CGAL). 

I might try installing OpenSCAD then playing with the [Windows search paths](https://github.com/mikedh/trimesh/blob/main/trimesh/interfaces/scad.py), maybe it needs to be case insensitive on windows or something?

**Comment by dhabbyc** [comment]
My english clearly is a problem here. 

 I can use blender/OpenSCAD through trimesh (both work on mi laptop). Nevertheless, I have other non-technical restrictions with both softwares (GPL licenses). Thats the reason I end up using pycork (LGPL).

 I'm not sure if there is a workaround to the GPL issue. As far as I understand, the use of, for instance, blender as an engine for the boolean operations is by means of a script (I looked at the trimesh source code), and they state (blender) that you can use scripts on a non GLP fashion... but I'm not sure.

 Thank you for your help.
